hsm-gateway-applications/scripts/generate-hsm-user.sh

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+set -eo pipefail
+
+# define the CA setup
+CA_HOST=localhost
+CA_URL=${CA_HOST}:7054
+TLS_CERT='../../test-network/organizations/fabric-ca/org1/tls-cert.pem'
+
+LocateHsmLib() {
+  if [[ -n "${PKCS11_LIB}" && -f "${PKCS11_LIB}" ]]; then
+    echo "${PKCS11_LIB}"
+    return
+  fi
+
+  local POSSIBLE_LIB_LOC=( \
+    '/usr/lib/softhsm/libsofthsm2.so' \
+    '/usr/lib/x86_64-linux-gnu/softhsm/libsofthsm2.so' \
+    '/usr/local/lib/softhsm/libsofthsm2.so' \
+    '/usr/lib/libacsp-pkcs11.so' \
+  )
+  for TEST_LIB in "${POSSIBLE_LIB_LOC[@]}"; do
+    if [ -f "${TEST_LIB}" ]; then
+      echo "${TEST_LIB}"
+      return
+    fi
+  done
+}
+
+HSM2_LIB=$(LocateHsmLib)
+[ -z $HSM2_LIB ] && echo No SoftHSM PKCS11 Library found, ensure you have installed softhsm2 && exit 1
```

Comment:
There are lots of unquoted variables that should probably be quoted to protect them from accidental globbing or word splitting. I recommend running [shellcheck](https://github.com/koalaman/shellcheck) against the script to pick up potential issues.
```suggestion
[ -z "${HSM2_LIB}" ] && echo 'No SoftHSM PKCS11 Library found, ensure you have installed softhsm2' && exit 1
```